sopel/modules/xkcd.py

```diff
@@ -53,6 +60,32 @@ def web_search(query):
         return match.group(1)
 
 
+def searchxkcd_search(query):
+    parameters = {
+        'q': query,
+        'page': 0,
+    }
+    try:
+        response = requests.post(SEARCHXKCD_API, params=parameters)
+    except requests.exceptions.ConnectionError as e:
+        LOGGER.debug("Unable to reach searchxkcd API: %s", e)
+        return None
+    except Exception as e:
+        LOGGER.debug("Unexpected error calling searchxkcd API: %s", e)
+        return None
+
+    try:
+        hits = response.json()['results']['hits']
+        if not hits:
+            return None
+        first = hits[0]['objectID']
+    except (JSONDecodeError, KeyError, IndexError):
+        LOGGER.debug("Data format from searchxkcd API could not be understood.")
```

Comment:
Hmm, the idea was that a failure here isn't really important enough to dump the whole exception, nor is it important enough to spit out an "Exception!" log entry, because of fallback to web-search.